src/main/java/com/microsoft/azure/relay/TokenRenewer.java

```diff
@@ -62,6 +61,7 @@ void onRenewTimer() {
 	}
 
 	private void scheduleRenewTimer(SecurityToken token) {
+		this.renewTimer = new Timer();
```

Comment:
>this.renewTimer = new Timer(); [](start = 2, length = 30)

If disconnect happens, TokenRenewer will close. Once the "renewTimer" object calls .cancel(), it will throw if we try to do anything with it. It can't just change to a new time like in C#, so I'm creating a new intance every time we schedule.